qemu/tests/block_hotplug_with_cpu_hotplug.py

```diff
@@ -0,0 +1,117 @@
+                             expected_num, test):
+        """
+        Check hotplug/unplug disks in guest.
+
+        :param session: The guest session object.
+        :param action: hotplug/unplug.
+        :param disks_before_plug: disks set before hotplug/unplug.
+        :param expcted_num: expected disks number in guest.
+        :param test: QEMU test object.
+        :return: changed disks set in guest.
+        """
+        if not utils_misc.wait_for(lambda: len(disks_before_plug ^ find_all_disks(
+               session)) == expected_num, 100, step=1.5):
+            disks_info_win = ('wmic logicaldisk get drivetype,name,description '
+                              '& wmic diskdrive list brief /format:list')
+            disks_info = session.cmd(disks_info_win)
+            test.log.info("The details of disks:\n %s", disks_info)
+            test.fail("Failed to {0} devices from guest, need to {0}: {1}, "
+                      "actual {0}: {2}".format(action, expected_num,
+                                               len(disks_before_plug ^ all_disks)))
+        return disks_before_plug ^ all_disks
+
+    def block_hotplug(image_name):
+        """
+        Hotplug disks and verify it in qtree.
+
+        :param image_name: Image name of hotplug disk
+        :return: List of objects for hotplug disk.
+        """
+        image_params = params.object_params(image_name)
+        devs = vm.devices.images_define_by_params(image_name,
+                                                  image_params, 'disk')
+        for dev in devs:
+            ret = vm.devices.simple_hotplug(dev, vm.monitor)
+            if ret[1] is False:
+                test.fail("Failed to hotplug device '%s'."
+                          "Output:\n%s" % (dev, ret[0]))
+        dtype = qdevices.QBlockdevNode if vm.check_capability(
+            Flags.BLOCKDEV) else qdevices.QDrive
+        devs = [dev for dev in devs if not isinstance(dev, dtype)]
+        return devs
+
+    vcpu_devices = params.objects("vcpu_devices")
+    expected_num = int(params["expected_num"])
+    action = params["action"]
+    img_name_list = params.get("images").split()
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+
+    if params.get_boolean("workaround_need"):
+        win_wora.modify_driver(params, session)
+
+    error_context.context("Plug device", test.log.info)
+    disks_before_plug = find_all_disks(session)
+    block_hotplug(img_name_list[1])
+    block_check_in_guest(session, action, disks_before_plug,
+                         expected_num, test)
+
+    for vcpu_dev in vcpu_devices:
+        error_context.context("Hotplug vcpu device: %s" % vcpu_dev,
+                              test.log.info)
+        vm.hotplug_vcpu_device(vcpu_dev)
+    if not utils_misc.wait_for(
+            lambda: cpu_utils.check_if_vm_vcpus_match_qemu(vm),
+            60):
+        test.fail("Actual number of guest CPUs is not equal to expected")
```

Comment:
@peixiu thanks for update
 update cfg for the win11 error as we don't have to test cpu hotplug with desktop.
For "Actual number and expected number value in fail log as well."  we call cpu_utils.py , it will print the error log 'avocado.test DEBUG| CPU quantity mismatched !!! guest said it got 39 but we assigned 40' to help us debug.
if add it to  " test.fail("Actual number of guest CPUs is not equal to expected")"  more code will needed, I think 
it is not  worth.

Reply:
It's ok for me, just an advice~